cpp/src/detail/shuffle_wrappers.cu

```diff
@@ -335,5 +335,103 @@ template rmm::device_uvector<size_t> groupby_and_count_edgelist_by_local_partiti
   std::optional<rmm::device_uvector<double>>& d_edgelist_weights,
   bool groupby_and_counts_local_partition);
 
+template <typename vertex_t, typename value_t, bool multi_gpu>
+void collect_vertex_values_to_local(raft::handle_t const& handle,
+                                    rmm::device_uvector<vertex_t>& d_vertices,
+                                    rmm::device_uvector<value_t>& d_values,
+                                    rmm::device_uvector<value_t>& d_local_values,
+                                    vertex_t local_vertex_first)
+{
+  if constexpr (multi_gpu) {
+    auto& comm           = handle.get_comms();
+    auto const comm_size = comm.get_size();
+
+    rmm::device_uvector<vertex_t> d_rx_vertices(0, handle.get_stream());
+    rmm::device_uvector<value_t> d_rx_values(0, handle.get_stream());
+
+    std::tie(d_rx_vertices, d_rx_values, std::ignore) =
+      cugraph::groupby_gpu_id_and_shuffle_kv_pairs(
+        comm,
+        d_vertices.begin(),
+        d_vertices.end(),
+        d_values.begin(),
+        [key_func = cugraph::detail::compute_gpu_id_from_vertex_t<vertex_t>{comm_size}] __device__(
```

Comment:
Have you checked with initial guess hub values? If I am not mistaken, you're calling this after renumbering, but `cugrpah::detail::compute_gpu_id_from_vertex_t` expects original (before renumbering) vertex IDs.

Reply:
adding a test in the next push.  Refactored this code as you suggested.